### DISCUSSION TITLE
howto build examples in Arduino IDE?

### DISCUSSION THREAD

**Discussion by artkoz** [tobozo/ESP32-USB-Soft-Host#36] [General]
Hello. Can you help more info for buiilding in Arduino IDE? Fresh install, esp32 support installed by boards manager.
error on building:
Arduino: 1.8.19 (Linux), Board: "ESP32 Wrover Module, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), QIO, 80MHz, 921600, None"
/home/user/Arduino/libraries/ESP32-USB-Soft-Host/src/usb_host.c:18:25: fatal error: hal/cpu_hal.h: No such file or directory
compilation terminated.
exit status 1
Error compiling for board ESP32 Wrover Module.

**Comment by tobozo** [comment]
Hi, thanks for your feedbak :+1: 

> esp32 support installed by boards manager.

Please check your package url Arduino Preferences (additional boards), it should be: 

https://espressif.github.io/arduino-esp32/package_esp32_index.json

>  hal/cpu_hal.h: No such file or directory

this is  a typical error when esp-idf version bundled with espressif core is too old, solution is to update the core to the latest version (2.0.9).

**Reply by artkoz** [reply]
oh i'm an idiot.. i was using the old ide install guide, with the wrong url
with the  2.0.9 -  build ok. 
thanks!
